webapp/tests/utils/channel_utils.test.jsx

```diff
@@ -0,0 +1,404 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import * as Utils from 'utils/channel_utils.jsx';
+import TeamStore from 'stores/team_store.jsx';
+import UserStore from 'stores/user_store.jsx';
+import ChannelStore from 'stores/channel_store.jsx';
+import Constants from 'utils/constants.jsx';
+
+describe('Channel Utils', () => {
+    describe('showDeleteOption', () => {
+        test('all users can delete channels on unlicensed instances', () => {
+            global.window.mm_license = {IsLicensed: 'false'};
+            expect(Utils.showDeleteOptionForCurrentUser(null)).
+                toEqual(true);
+        });
+
+        test('users cannot delete default channels', () => {
+            global.window.mm_license = {IsLicensed: 'true'};
+            const channel = {name: Constants.DEFAULT_CHANNEL};
+            expect(Utils.showDeleteOptionForCurrentUser(channel)).
+                toEqual(false);
+        });
+
+        test('system admins can delete private channels, user is system admin test', () => {
+            global.window.mm_license = {IsLicensed: 'true'};
+            global.window.mm_config = {RestrictPrivateChannelDeletion: Constants.PERMISSIONS_SYSTEM_ADMIN};
+            UserStore.isSystemAdminForCurrentUser = () => {
+                return true;
+            };
```

Comment:
I *think* this is OK, but I'm not certain. There are other tests already that do stuff to the global config and that seems to work OK.